custom_components/localtuya/fan.py

```diff
@@ -74,15 +75,22 @@ def turn_off(self, **kwargs) -> None:
 
     def set_speed(self, speed: str) -> None:
         """Set the speed of the fan."""
+        mappings = {
```

Comment:
I do feel that this mapping thing (if reported state is a number, use a number otherwise use a string) is kinda magic and hides what is happening. I understand that auto-detection is nice and all, but it clutters the code and makes it harder to understand. Especially later, when we get another special case. And another. And another... A pre-defined list of values would be better, like we do for open/close commands in `cover`. But we might as well leave it like this now and fix that whenever we encounter another set of speeds.